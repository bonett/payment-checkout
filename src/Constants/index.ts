export const RESPONSE_SERVER = [
  {
    id: "4483b01e-3c83-4f5a-8e26-65c419d24570",
    title: "black boots, 100% original leather",
    price: 213131,
    currency_id: "COP",
    thumbnail:
      "https://img.freepik.com/free-photo/travel-still-life-pack-top-view_23-2148837322.jpg?w=1800&t=st=1705363387~exp=1705363987~hmac=6e3c55c14a115cde11830ae5b8babaf38e22f4974aff67fa809b2a7efe449e62",
    description:
      "Magna eiusmod proident non culpa et in sunt consectetur esse amet sit Enim et exercitation mollit duis sit elit exercitation non magna Ipsum sint commodo laboris non adipisicing ad dolor aliquip est voluptate ut Aute enim dolore aute non Tempor magna excepteur ut sit Lorem irure deserunt in proident",
  },
  {
    id: "e9156f7d-c780-4333-bf8c-06346c7dc773",
    title: "Male travel objects for different places",
    price: 1122312,
    currency_id: "COP",
    thumbnail:
      "https://img.freepik.com/free-photo/male-travel-objects_23-2147635500.jpg?w=1800&t=st=1705363805~exp=1705364405~hmac=2818d460f3067c8ce52996a8f69ade4c3c31a95a7c44a2e7f4836e711ea78b7e",
    description:
      "Voluptate mollit ea occaecat magna tempor eu sunt cupidatat enim cupidatat minim magna tempor pariatur. Quis velit proident in duis velit pariatur amet adipisicing veniam voluptate non aliquip non in.",
  },
  {
    id: "f622b447-2bf8-47f0-b1c7-f91a3420e53e",
    title: "Flat lay composition for men, recommended for traveling",
    price: 2343242,
    currency_id: "COP",
    thumbnail:
      "https://img.freepik.com/free-photo/flat-lay-composition-different-traveling-elements_23-2148884945.jpg?w=1800&t=st=1705363395~exp=1705363995~hmac=bab6028d8dfcd8d94b3e90e584d2256c95bec430b37a1780c88fec81993b600a",
    description:
      "Voluptate mollit ea occaecat magna tempor eu sunt cupidatat enim cupidatat minim magna tempor pariatur. Quis velit proident in duis velit pariatur amet adipisicing veniam voluptate non aliquip non in.",
  },
  {
    id: "6541a7be-3ea5-4731-bb15-295a360c5345",
    title: "Travel still life pack for men, leather accesories",
    price: 2423,
    currency_id: "COP",
    thumbnail:
      "https://img.freepik.com/free-photo/travel-still-life-pack-flat-lay_23-2148837355.jpg?w=1800&t=st=1705363406~exp=1705364006~hmac=d3406e925f1bb6bc97688072b2c71f7bee50f494bc204b525866e7748878180d",
    description:
      "Voluptate mollit ea occaecat magna tempor eu sunt cupidatat enim cupidatat minim magna tempor pariatur. Quis velit proident in duis velit pariatur amet adipisicing veniam voluptate non aliquip non in.",
  },
  {
    id: "0eb53654-796a-4949-ae0a-ada166715fc0",
    title: "Laptop with blank screen",
    price: 932567,
    currency_id: "COP",
    thumbnail:
      "https://img.freepik.com/free-photo/laptop-with-blank-black-screen-white-table_53876-97915.jpg?w=2000&t=st=1705388096~exp=1705388696~hmac=0a702ff125bc27152bafeb09db0b31a6fb6d511e382d9cfc9330992efb20cc72",
    description:
      "Magna eiusmod proident non culpa et in sunt consectetur esse amet sit. Enim et exercitation mollit duis sit elit exercitation non magna. Ipsum sint commodo laboris non adipisicing ad dolor aliquip est voluptate ut. Aute enim dolore aute non. Tempor magna excepteur ut sit Lorem irure deserunt in proident.",
  },
  {
    id: "558e0996-c707-49b2-9299-7567ed95ba89",
    title: "Iphone with blue background balancing - 1T",
    price: 6898,
    currency_id: "COP",
    thumbnail:
      "https://img.freepik.com/free-photo/smartphones-balancing-with-blue-background_23-2150271748.jpg?w=1480&t=st=1705388169~exp=1705388769~hmac=93447f79811d2363d8e3a053e314ccd5f45dcb242915737d96b483816e56aae1",
    description:
      "Voluptate mollit ea occaecat magna tempor eu sunt cupidatat enim cupidatat minim magna tempor pariatur. Quis velit proident in duis velit pariatur amet adipisicing veniam voluptate non aliquip non in.",
  },
  {
    id: "23582fe2-bb74-481c-ae51-77ee5f2b0fc9",
    title: "Desk with gray laptop chair",
    price: 932567,
    currency_id: "COP",
    thumbnail:
      "https://img.freepik.com/free-photo/desk-with-gray-laptop-chair_23-2148267493.jpg?w=2000&t=st=1705388253~exp=1705388853~hmac=be013b1ad0ede3500b641184285e4bee8f814312ed90362315bf74f76112781a",
    description:
      "Magna eiusmod proident non culpa et in sunt consectetur esse amet sit. Enim et exercitation mollit duis sit elit exercitation non magna. Ipsum sint commodo laboris non adipisicing ad dolor aliquip est voluptate ut. Aute enim dolore aute non. Tempor magna excepteur ut sit Lorem irure deserunt in proident.",
  },
  {
    id: "c2c36d91-8fb1-47b0-9f00-80d8ff54c26c",
    title: "Excel version 12, compatible with opera",
    price: 2342,
    currency_id: "COP",
    thumbnail:
      "https://img.freepik.com/free-photo/spreadsheet-document-information-financial-startup-concept_53876-124145.jpg?w=2000&t=st=1705388333~exp=1705388933~hmac=9b7c1bc086d297dbb3108b17d872db7cc697341090ce94d96f6cacf928d8d5bc",
    description:
      "Aute enim dolore aute non. Tempor magna excepteur ut sit Lorem irure deserunt in proident. Voluptate mollit ea occaecat magna tempor eu sunt cupidatat enim cupidatat minim magna tempor pariatur. Quis velit proident in duis velit pariatur amet adipisicing veniam voluptate non aliquip non in.",
  },
];

export const TAXES = 712836;

export const USER_INITIAL_STATE = {
  formStep: 1,
  isOpenModal: false,
  formData: {
    number: "",
    expiry: "",
    cvc: "",
    name: "",
    focus: "",
  },
  productSelected: null,
  responsePayment: null,
};
