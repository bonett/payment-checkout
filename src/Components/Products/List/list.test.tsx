import React from "react";
import { render, screen } from "@testing-library/react";
import ProductList from "./";
import { RESPONSE_SERVER } from "Constants";
import { Provider } from "react-redux";
import store from "Store";

describe("Product List component", () => {
  test("Should render ProductList with 8 cards", () => {
    const { container } = render(
      <Provider store={store}>
        <ProductList products={RESPONSE_SERVER} />
      </Provider>
    );

    const cards = screen.getAllByTestId("product-card");
    expect(cards.length).toEqual(8);
    expect(container).toMatchSnapshot();
  });
});
