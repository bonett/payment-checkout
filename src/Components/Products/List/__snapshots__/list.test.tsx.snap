// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product List component Should render ProductList with 8 cards 1`] = `
<div>
  <div
    class="product-content"
  >
    <article
      class="article-content"
      data-testid="product-card"
    >
      <div
        class="article-content_thumb"
      >
        <img
          alt="black boots, 100% original leather"
          src="https://img.freepik.com/free-photo/travel-still-life-pack-top-view_23-2148837322.jpg?w=1800&t=st=1705363387~exp=1705363987~hmac=6e3c55c14a115cde11830ae5b8babaf38e22f4974aff67fa809b2a7efe449e62"
        />
      </div>
      <div
        class="article-content_description"
      >
        <h2
          class="article-content_title"
        >
          black boots, 100% original leather
        </h2>
        <p
          class="article-content_parragraph"
        >
          Magna eiusmod proident non culpa et in sunt consectetur esse amet sit Enim et exercitation mollit duis sit elit exercitation non magna Ipsum sint commodo laboris non adipisicing ad dolor aliquip est voluptate ut Aute enim dolore aute non Tempor magna excepteur ut sit Lorem irure deserunt in proident
        </p>
        <div
          class="article-content_action"
        >
          <h6
            class="article-content_price"
          >
            $ 213.131
          </h6>
          <button
            class="btn-primary"
          >
            Buy
          </button>
        </div>
      </div>
    </article>
    <article
      class="article-content"
      data-testid="product-card"
    >
      <div
        class="article-content_thumb"
      >
        <img
          alt="Male travel objects for different places"
          src="https://img.freepik.com/free-photo/male-travel-objects_23-2147635500.jpg?w=1800&t=st=1705363805~exp=1705364405~hmac=2818d460f3067c8ce52996a8f69ade4c3c31a95a7c44a2e7f4836e711ea78b7e"
        />
      </div>
      <div
        class="article-content_description"
      >
        <h2
          class="article-content_title"
        >
          Male travel objects for different places
        </h2>
        <p
          class="article-content_parragraph"
        >
          Voluptate mollit ea occaecat magna tempor eu sunt cupidatat enim cupidatat minim magna tempor pariatur. Quis velit proident in duis velit pariatur amet adipisicing veniam voluptate non aliquip non in.
        </p>
        <div
          class="article-content_action"
        >
          <h6
            class="article-content_price"
          >
            $ 1.122.312
          </h6>
          <button
            class="btn-primary"
          >
            Buy
          </button>
        </div>
      </div>
    </article>
    <article
      class="article-content"
      data-testid="product-card"
    >
      <div
        class="article-content_thumb"
      >
        <img
          alt="Flat lay composition for men, recommended for traveling"
          src="https://img.freepik.com/free-photo/flat-lay-composition-different-traveling-elements_23-2148884945.jpg?w=1800&t=st=1705363395~exp=1705363995~hmac=bab6028d8dfcd8d94b3e90e584d2256c95bec430b37a1780c88fec81993b600a"
        />
      </div>
      <div
        class="article-content_description"
      >
        <h2
          class="article-content_title"
        >
          Flat lay composition for men, recommended for traveling
        </h2>
        <p
          class="article-content_parragraph"
        >
          Voluptate mollit ea occaecat magna tempor eu sunt cupidatat enim cupidatat minim magna tempor pariatur. Quis velit proident in duis velit pariatur amet adipisicing veniam voluptate non aliquip non in.
        </p>
        <div
          class="article-content_action"
        >
          <h6
            class="article-content_price"
          >
            $ 2.343.242
          </h6>
          <button
            class="btn-primary"
          >
            Buy
          </button>
        </div>
      </div>
    </article>
    <article
      class="article-content"
      data-testid="product-card"
    >
      <div
        class="article-content_thumb"
      >
        <img
          alt="Travel still life pack for men, leather accesories"
          src="https://img.freepik.com/free-photo/travel-still-life-pack-flat-lay_23-2148837355.jpg?w=1800&t=st=1705363406~exp=1705364006~hmac=d3406e925f1bb6bc97688072b2c71f7bee50f494bc204b525866e7748878180d"
        />
      </div>
      <div
        class="article-content_description"
      >
        <h2
          class="article-content_title"
        >
          Travel still life pack for men, leather accesories
        </h2>
        <p
          class="article-content_parragraph"
        >
          Voluptate mollit ea occaecat magna tempor eu sunt cupidatat enim cupidatat minim magna tempor pariatur. Quis velit proident in duis velit pariatur amet adipisicing veniam voluptate non aliquip non in.
        </p>
        <div
          class="article-content_action"
        >
          <h6
            class="article-content_price"
          >
            $ 2.423
          </h6>
          <button
            class="btn-primary"
          >
            Buy
          </button>
        </div>
      </div>
    </article>
    <article
      class="article-content"
      data-testid="product-card"
    >
      <div
        class="article-content_thumb"
      >
        <img
          alt="Laptop with blank screen"
          src="https://img.freepik.com/free-photo/laptop-with-blank-black-screen-white-table_53876-97915.jpg?w=2000&t=st=1705388096~exp=1705388696~hmac=0a702ff125bc27152bafeb09db0b31a6fb6d511e382d9cfc9330992efb20cc72"
        />
      </div>
      <div
        class="article-content_description"
      >
        <h2
          class="article-content_title"
        >
          Laptop with blank screen
        </h2>
        <p
          class="article-content_parragraph"
        >
          Magna eiusmod proident non culpa et in sunt consectetur esse amet sit. Enim et exercitation mollit duis sit elit exercitation non magna. Ipsum sint commodo laboris non adipisicing ad dolor aliquip est voluptate ut. Aute enim dolore aute non. Tempor magna excepteur ut sit Lorem irure deserunt in proident.
        </p>
        <div
          class="article-content_action"
        >
          <h6
            class="article-content_price"
          >
            $ 932.567
          </h6>
          <button
            class="btn-primary"
          >
            Buy
          </button>
        </div>
      </div>
    </article>
    <article
      class="article-content"
      data-testid="product-card"
    >
      <div
        class="article-content_thumb"
      >
        <img
          alt="Iphone with blue background balancing - 1T"
          src="https://img.freepik.com/free-photo/smartphones-balancing-with-blue-background_23-2150271748.jpg?w=1480&t=st=1705388169~exp=1705388769~hmac=93447f79811d2363d8e3a053e314ccd5f45dcb242915737d96b483816e56aae1"
        />
      </div>
      <div
        class="article-content_description"
      >
        <h2
          class="article-content_title"
        >
          Iphone with blue background balancing - 1T
        </h2>
        <p
          class="article-content_parragraph"
        >
          Voluptate mollit ea occaecat magna tempor eu sunt cupidatat enim cupidatat minim magna tempor pariatur. Quis velit proident in duis velit pariatur amet adipisicing veniam voluptate non aliquip non in.
        </p>
        <div
          class="article-content_action"
        >
          <h6
            class="article-content_price"
          >
            $ 6.898
          </h6>
          <button
            class="btn-primary"
          >
            Buy
          </button>
        </div>
      </div>
    </article>
    <article
      class="article-content"
      data-testid="product-card"
    >
      <div
        class="article-content_thumb"
      >
        <img
          alt="Desk with gray laptop chair"
          src="https://img.freepik.com/free-photo/desk-with-gray-laptop-chair_23-2148267493.jpg?w=2000&t=st=1705388253~exp=1705388853~hmac=be013b1ad0ede3500b641184285e4bee8f814312ed90362315bf74f76112781a"
        />
      </div>
      <div
        class="article-content_description"
      >
        <h2
          class="article-content_title"
        >
          Desk with gray laptop chair
        </h2>
        <p
          class="article-content_parragraph"
        >
          Magna eiusmod proident non culpa et in sunt consectetur esse amet sit. Enim et exercitation mollit duis sit elit exercitation non magna. Ipsum sint commodo laboris non adipisicing ad dolor aliquip est voluptate ut. Aute enim dolore aute non. Tempor magna excepteur ut sit Lorem irure deserunt in proident.
        </p>
        <div
          class="article-content_action"
        >
          <h6
            class="article-content_price"
          >
            $ 932.567
          </h6>
          <button
            class="btn-primary"
          >
            Buy
          </button>
        </div>
      </div>
    </article>
    <article
      class="article-content"
      data-testid="product-card"
    >
      <div
        class="article-content_thumb"
      >
        <img
          alt="Excel version 12, compatible with opera"
          src="https://img.freepik.com/free-photo/spreadsheet-document-information-financial-startup-concept_53876-124145.jpg?w=2000&t=st=1705388333~exp=1705388933~hmac=9b7c1bc086d297dbb3108b17d872db7cc697341090ce94d96f6cacf928d8d5bc"
        />
      </div>
      <div
        class="article-content_description"
      >
        <h2
          class="article-content_title"
        >
          Excel version 12, compatible with opera
        </h2>
        <p
          class="article-content_parragraph"
        >
          Aute enim dolore aute non. Tempor magna excepteur ut sit Lorem irure deserunt in proident. Voluptate mollit ea occaecat magna tempor eu sunt cupidatat enim cupidatat minim magna tempor pariatur. Quis velit proident in duis velit pariatur amet adipisicing veniam voluptate non aliquip non in.
        </p>
        <div
          class="article-content_action"
        >
          <h6
            class="article-content_price"
          >
            $ 2.342
          </h6>
          <button
            class="btn-primary"
          >
            Buy
          </button>
        </div>
      </div>
    </article>
  </div>
</div>
`;
